### ISSUE TITLE
Is there a way to do a "contain rule" on the filter_by negation ?

### ISSUE THREAD

**Issue by tahola** [typesense/typesense#1264]
Hello,

`filter_by=genre:!=[Drama,Action]`

this feature is very useful but let say that my "tags" are more precise like `Action Adventure` or `Action Comedy` and I want to remove all the results with an Action genre, is there a way to do that ?

I mean, something like `filter_by=genre:!=[Action*]` or `filter_by=genre:!=[Action^]`

**Comment by kishorenc** [comment]
We don't have a way to do this, but we probably should support a `genre:! [Action]` syntax (without the `=`).

**Comment by tahola** [comment]
I am sorry I dont get you, I the doc I see the = 

> Negation:
> Not equals / negation is supported via the :!= operator, e.g. filter_by=author:!=JK Rowling. You can also negate multiple values: filter_by=author:!=[JK Rowling, Gilbert Patten]

**Comment by kishorenc** [comment]
Yes, we don't suppose this yet. I've marked this issue as a feature request.

**Comment by aygunia** [comment]
If we could add support for this that would be great. 

Essentially typesense right now supports `WHERE LIKE X` but **not** `WHERE NOT LIKE X`, and the existing negation only supports an exact `WHERE !=`.

For example, the WHERE LIKE works but the WHERE NOT LIKE does not, and if I follow the documentation the exact negation with “:!=“ doesn’t work for me, as I don’t have the exact value.

Some values I have stored below, I want to filter out any entries that contain `STOCK` which is only possible with a `WHERE NOT LIKE`, I can include `STOCK` with `variant_title:-STOCK-` but I cannot exclude with what I would assume would be `variant_title:!-STOCK-`

`UK 7.5 | EU 42 | US 10`
`UK 8 | EU 43 | US 11-STOCK-142`

**Comment by kishorenc** [comment]
Yes, we will be taking this up for implementation in the next couple of weeks.

**Comment by rdsedmundo** [comment]
Would this new operator work with complex filters, like `! (status:active || n_rec:4)`? i.e. having all the flexibility of the `NOT` from SQL.

**Comment by kishorenc** [comment]
We don't support the NOT operator on complex filters yet.
